src/plugins/gs-plugin-eos.c

```diff
@@ -401,6 +460,38 @@ gs_plugin_adopt_app (GsPlugin *plugin, GsApp *app)
 	gs_app_set_management_plugin (app, gs_plugin_get_name (plugin));
 }
 
+static void
+gs_plugin_eos_refine_app_branch (GsPlugin *plugin, GsApp *app)
+{
+	const char *default_branch;
+	const char *branch = NULL;
+	const char *origin = gs_app_get_origin (app);
+	GsPluginData *priv = NULL;
+
+	if (!gs_plugin_eos_app_is_flatpak (app))
+		return;
+
+	priv = gs_plugin_get_data (plugin);
+	default_branch = (char *) g_hash_table_lookup (priv->default_branches,
+						       origin);
+
+	/* if we do not have a configured default branch for this repo then
+	 * do nothing */
+	if (!default_branch)
+		return;
+
+	/* if an app has no branch set, maybe it will be set later so we let
+	 * it pass */
+	branch = gs_app_get_flatpak_branch (app);
+	if (!branch)
+		return;
+
+	/* do not show an app if it doesn't belong to the default branch that
+	 * is configured for its remote */
+	if (branch && g_strcmp0 (branch , default_branch) != 0)
```

Comment:
`default_branch` can't be `NULL` here (you check above), and `g_strcmp0()` will protect you from `branch` being `NULL`, so you can remove the `branch &&` part of this